### PR TITLE
feat: add jina deployment support

### PR DESCRIPTION
**Pull request by subbuv26**
**Goal**
--------
Add support for all the below operations for Jina Deployment
- [x] list
- [x] remove
- [x] update
- [x] restart
- [x] pause
- [x] resume
- [x] scale
- [x] recreate
- [x] status
- [x] deploy
- [x] logs

----------------

- [x] Run [Integration tests GHA](https://github.com/jina-ai/jcloud/actions/workflows/integration-tests.yml) manually & comment the link. 👉 https://github.com/jina-ai/jcloud/actions/runs/6508262154

@jina-ai/team-wolf

### CODE REVIEW

File: jcloud/deployment.py [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+)
+
+logger = get_logger()
+
+pbar, pb_task = get_pbar(
+    '', total=2, disable='JCLOUD_NO_PROGRESSBAR' in os.environ
+)  # progress bar for deployment
+
+
+def _exit_if_response_error(
```
Comment (npitsillos) [inline jcloud/deployment.py:45]: This also exists in `jcloud/flow.py` can we remove this and use a generic one?

File: jcloud/deployment.py [hunk 2]
```diff
@@ -0,0 +1,634 @@
...
+            print_server_response(json_response['error'])
+            exit_error(f'Please make sure the requested resource exists.')
+        else:
+            exit_error(
+                f'Bad response: expecting [b]{expected_status}[/b], got [b]{response.status}[/b] from server.\n'
+                f'{json.dumps(json_response, indent=1)}'
+            )
+
+
+def print_server_response(error_message: str):
```
Comment (npitsillos) [inline jcloud/deployment.py:68]: Same as above

**Review by npitsillos — CHANGES_REQUESTED** [review]
Some minor comments on using a generic function for when an error occurs.

**Review by npitsillos — APPROVED** [review]

**Review by tarrantro — APPROVED** [review]
lgtm

File: jcloud/parsers/update.py [hunk 3]
```diff
@@ -48,6 +68,19 @@ def _set_update_secret_parser(update_parser):
         help='The string ID of the Flow.',
     )
 
+    # TODO (subbu) need make flow to be optional as deployment can be used instead of flow
```
Comment (tarrantro) [inline jcloud/parsers/update.py:71]: what is deployment can be used instead of flow?

**Pull request merged**